### DISCUSSION TITLE
Activity Monitor context menu option

### DISCUSSION THREAD

**Discussion by lemoran** [lawand-dot-io/taskbar#35] [Feature requests]
That's it,  like Task Manager in Windows. This option can be placed in the taskbar context menu. And can also be present in the "Taskbar" submenu that shows when right clicking a Window button on taskbar.
![taskbar-context-menu](https://github.com/lawand-dot-io/taskbar/assets/2117089/d1d400fa-d4ac-4f4a-8cde-3fb8ed6d2d82)
![window-button-context-menu](https://github.com/lawand-dot-io/taskbar/assets/2117089/1b05672c-4892-4128-89d8-3fb3872fc72b)

**Comment by lawand-dot-io** [comment]
Hello. I guess I could add this. I feel like it will clutter the context menu, so maybe I would make it disabled by default and I would add an option in the preferences to enable showing the activity monitor. Until I implement this, feel free to select the "Keep in Taskbar" option for the activity monitor as a temporary solution if you use it frequently. Cheers

**Comment by lemoran** [comment]
Hi! Yes, you are right, I feel the same. May be too much items in the menu, and since here we are talking about a Windows style taskbar, but Mac users don't like the cluttered experience of Windows so I agree keeping it simple will be better.
From the simplicity point of view, it can be an optional item (thru preferences), and may be there are also two candidates for having a preferences option (checkbox) to show them (it's my personal opinion, just take a look if it sounds good to you):
- The taskbar submenu
- The Add to blacklist item (I think most people will not use it)
<img width="363" alt="Screenshot 2024-03-16 at 12 14 31" src="https://github.com/lawand-dot-io/taskbar/assets/2117089/5cdf0139-6904-49aa-8768-305c17b0ee13">
<img width="220" alt="Screenshot 2024-03-16 at 12 14 39" src="https://github.com/lawand-dot-io/taskbar/assets/2117089/cdb1aaee-e852-4bb2-9356-dc6c43b5f068">

EDIT: the "Add to blacklist" item can also be confusing for some users that don't have enough experience to get into Preferences and remove an accidentally added application to that list (Someone can accidentally click on the option).

**Reply by lawand-dot-io** [reply]
great ideas. I never thought about adding an option to customize the context menus, but you're right, especially when I start adding more and more options there. also, explaining stuff like blacklist makes sense. I should probably show a popup that explains what is about to happen and allow cancelling and "don't show this again" as a checkbox. thanks for the feedback and feel free to reach out again with any ideas. cheers

**Reply by lemoran** [reply]
Hi, I was about to open another discussion but I just realized it does not make sense, since it's little relevance detail.
In the same line of decluttering and making design simpler, I just realized another detail about the user interface: tray icon menu has a disabled item called "Taskbar" that does nothing since it's always disabled. I guess you put that item to identify that the user clicked Taskbar menu and not other.
<img width="236" alt="1" src="https://github.com/lawand-dot-io/taskbar/assets/2117089/96ef5975-f84d-4148-838c-3d1138f180c4">
I think it can be enhanced by removing that item and the following menu separator (the horizontal line), and just modifying the "About" item to "About Taskbar". With that simple change the user can identify he/she clicked the Taskbar menu (similar to AltTab menu).
<img width="246" alt="2" src="https://github.com/lawand-dot-io/taskbar/assets/2117089/53a2690f-bc7c-4816-b938-b6f06326ea4a">
It's just a suggestion of course! But I hope it can help!

**Reply by lawand-dot-io** [reply]
@lemoran Again thanks for the ideas :) I guess it's true the the greyed out Taskbar item might be confusing. I will keep in mind to probably change About to About Taskbar. Thanks and I hope you're enjoying Taskbar. Cheers

**Comment by livinlicious** [comment]
I feel this is already possible.
By "keep in taskbar" the activity monitor:
https://monosnap.com/file/gfj42pT74a3tQSuyhXnPHnDmnAmuRZ

This just opens it up. It might be a quick and dirty fix for your issue.
